source/logging/logging.rst

```diff
@@ -0,0 +1,380 @@
+==============================
+Client Side Operations Timeout
+==============================
+
+:Title: Logging
+:Status: Accepted
+:Minimum Server Version: N/A
+
+.. contents::
+
+--------
+
+Abstract
+========
+This specification defines requirements for drivers' logging configuration and behavior. 
+
+
+META
+====
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”,
+“SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this
+document are to be interpreted as described in `RFC 2119
+<https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+-----
+Terms
+-----
+Structured logging
+    Structured logging refers to producing log messages in a structured format, i.e. a series of
+    key-value pairs, which can be converted to external formats such as JSON.
+
+Unstructured logging
+    Unstructured logging refers to producing string log messages which embed all attached
+    information within that string.
+
+---------------------------
+Implementation requirements
+---------------------------
+Drivers SHOULD implement support for logging in a manner that is idiomatic for their language and
+ecosystem.
+
+
+Minimizing Required Code Changes
+--------------------------------
+If possible, drivers SHOULD support the following configuration requirements in a manner that does
+not require any changes to an application's source code, and in the case of compiled languages,
+does not require recompilation.
+However, if that is impossible to do without conflicting with idiomatic logging patterns for the
+language ecosystem, it is acceptable for a driver to require minimal changes to enable logging,
+such as recompiling with a feature flag specified, or passing a logging configuration to a 
+``MongoClient`` constructor. In that case, drivers SHOULD strive to minimize the amount of code
+change needed.
+
+Drivers SHOULD take advantage of built-in support for logging configuration within logging
```

Comment:
the "fallback implementation methods" here are all things I have not actually attempted to do, because Rust has a logging framework to rely on.
I expect we might have to adjust these once a language without a logging framework actually attempts this (iirc @jmikola is going to do the libmongoc implementation later this quarter?)